Log analytic events for CC Libraries

shaoshing opened the pull request:
Here are the events logged:

* create/delete/update (only rename)/switch library
* create/delete/update (only rename) element
* apply color / color-theme / layerstyle / characterstyle
* place graphic
* open/update graphic
* open library links: share library, share public link, view online
* open element links: share public link, view online
* open Adobe Stock link

iwehrman commented on src/js/actions/libraries.js:
```diff
@@ -1199,6 +1200,36 @@ define(function (require, exports) {
+                            headlights.logEvent("libraries", "library", "delete");
+                            break;
+                        case "createElement":
+                            // 
```
Missing helpful comment here? Or stray, stale comment?

iwehrman commented on src/js/util/headlights.js:
```diff
@@ -44,6 +45,7 @@ define(function (require, exports) {
      */
     var logEvent = function (category, subcategory, event) {
         if (global.debug) {
+            log.info("headlights: %s > %s > %s", category, subcategory, event);
```
I think this should be removed before this is merged. I know it only happens in debug mode only, but I'm still interested in keep the console output readable for developers, which means trying to limit the output to that which is widely useful. I think, outside of testing this PR, this probably won't be widely useful.

iwehrman commented on src/js/jsx/sections/libraries/LibraryList.jsx:
```diff
@@ -253,6 +254,20 @@ define(function (require, exports, module) {
+         *
+         * @private
+         */
+        _handleLibraryButtonClicked: function (event, url) {
```
Missing `@param` annotations.

iwehrman commented on src/js/jsx/sections/libraries/assets/AssetSection.jsx:
```diff
@@ -162,6 +161,20 @@ define(function (require, exports, module) {
+         *
+         * @private
+         */
+        _handleElementButtonClicked: function (event, url) {
```
Missing `@param` annotations.

iwehrman commented on src/js/jsx/sections/libraries/LibraryBar.jsx:
```diff
@@ -169,14 +169,28 @@ define(function (require, exports, module) {
+         *
+         * @private
+         */
+        _handleStockButtonClicked: function () {
```
Missing `@param` annotations.

iwehrman commented:
Fantastic! Just a few minor changes suggested. We'll get this merged before the standup. Back to @shaoshing.

iwehrman commented:
Also, just want to confirm that you explicitly opted not to use `CCLibraries.analyticsUtils.reportUseElement` for a good reason. Like, e.g., it doesn't actually serve any purpose?

shaoshing commented:
@iwehrman thanks for reviewing. I've addressed your comments. I actually missed the `reportUseElement` util function; but my first looks shows it is similar to using the `headlights.logEvent`, and require more codes to emit the events and then handle them together in the `reportEvent` callback, so probably not worth a lot to switch to it. Please let me know if you disagree.

iwehrman commented:
Based on my limited understanding, I don't disagree with your decision to skip `reportUseElement`.
